torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2568,29 +2572,27 @@ void ProcessGroupNCCL::broadcastUniqueNCCLID(
   }
 }
 
-void ProcessGroupNCCL::allgatherUniqueNCCLID(
+// We want to all-gather unique NCCL IDs from all roots using TCPStore.
+// This is first done by setting the ID by each root and then `multiGet` by all
+// ranks.
+void ProcessGroupNCCL::allgatherUniqueNCCLIDs(
     int rootRank,
+    int rootIdx,
     ncclUniqueId* ncclID,
     std::vector<ncclUniqueId>& ncclIDs) {
   std::vector<std::string> storeKeys;
+  std::vector<std::vector<uint8_t>> results;
   for (size_t r = 0; r < ncclIDs.size(); r++) {
     storeKeys.emplace_back("UniqueNCCLID:" + std::to_string(r));
   }
   if (rank_ == rootRank) {
```

Comment:
I was wondering, which `rootRank` do you pass to each rank, for this if testing.

Reply:
the rootRank belongs to that specific rank.

Reply:
For the example 0 - 10 and 3 nodes.

root rank for [0, 3] is 0
root rank for [4, 7] is 4
root rank for [8, 10] is 8.

And rank_ here is just the global rank.